inspirehep/utils/url.py

```diff
@@ -65,3 +69,17 @@ def is_pdf_link(url):
     correct_magic_number = magic_number.startswith('%PDF')
 
     return correct_magic_number
+
+
+def retrieve_uri(uri, outdir=None):
+    """Retrieves the given uri and stores it in a temporary file."""
+    fd, local_file = tempfile.mkstemp(prefix='inspire', dir=outdir)
+    os.close(fd)
+
+    with fsopen(uri, 'rb') as remote_fd:
+        data = remote_fd.read()
```

Comment:
This is not safe. It must be bufferized.

Reply:
I was not expecting to have big files (GB order), but sure, better safe than sorry.

Reply:
There might be hundredths of MB indeed. And also we might have some malicious user pointing to something huge...

OK, very improbable, but given the small size of our VMs... :-)